app/src/main/java/com/example/benchproject/data/movie/details/repo/MovieDetailsDataRepository.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.benchproject.data.movie.details.repo
+
+import com.example.benchproject.data.movie.details.entity.MovieDetailsRemote
+import com.example.benchproject.data.movies.common.api.MoviesApi
+import com.example.benchproject.data.movies.common.constants.MoviesConstants.Companion.BASE_URL_FOR_IMAGE
+import com.example.benchproject.domain.movie.details.entity.GenresModel
+import com.example.benchproject.domain.movie.details.entity.MovieDetailsModel
+import com.example.benchproject.domain.movie.details.repo.MovieDetailsRepository
+import javax.inject.Inject
+
+class MovieDetailsDataRepository @Inject constructor(
+    private val apiService: MoviesApi
+) : MovieDetailsRepository {
+    override suspend fun getMovieDetails(movieId: Int): Result<MovieDetailsModel> =
+        runCatching {
+            mapToMovieDetailsModelType(apiService.getMovieDetails(movieId))
+        }
+
+    private fun mapToMovieDetailsModelType(movieDetailsRemote: MovieDetailsRemote) =
```

Comment:
It should be an extension function.

Reply:
I changed this function to an extension function for `MovieDetailsRemote`